prometheus.lua

```diff
@@ -261,6 +261,39 @@ local function construct_bucket_format(buckets)
   return "%0" .. (max_order + max_precision + 1) .. "." .. max_precision .. "f"
 end
 
+-- Format bucket format when expose metrics.
+--
+-- This receives a key, remove leading and trailing zeros and return the cleaner
+-- number, in order to make the output more clear, without effect on the increasing
+-- order.
+--
+-- Args:
+--   key: the metric key
+--
+-- Returns:
+--   (string) the formatted key
+local function format_bucket_when_expose(key)
+  local all_le_string = string.match(key, '(le=".*")')
+  local part1, bucket, part2 = string.match(key, '(le=")(.*)(")')
+  if tonumber(bucket) == nil then
+    return key
+  end
+
+  --remove leading zeros
+  local _
+  _, bucket = string.match(bucket, '(0*)(.*)')
```

Comment:
This will remove all leading zeros (e.g. `00.01` -> `.01`) you'd probably want to leave one before decimal point (`0.01`).

Actually, I think the simplest way to do the number formatting here would be `bucket = tostring(tonumber(bucket))`. This will take care of both leading and trailing zeros. I'm not 100% sure about speed, but I think it won't be bad at all, because both functions are probably implemented in C and will be faster than using patterns and iterating strings in Lua.

Reply:
Yeah that's right! That is faster (regex in pure lua is indeed pretty slow) and way more elegant. Thanks a lot!